scripts/test-end-to-end.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
```

Comment:
I find shell scripts super annoying to debug. So I think we should always have the following flags enabled:

```
set -o errexit
set -o pipefail
```

docs are here: https://www.gnu.org/software/bash/manual/bash.html#The-Set-Builtin